docs/source/userguide/newton_solver.rst

```diff
@@ -0,0 +1,90 @@
+
+The parameters of the Newton solver for :class:`festim.HTransportProblem` can be chosen in :class:`festim.Settings` (see :ref:`settings_ug` for an example). Absolute and relative tolerances of the Newton solver
+are defined with ``absolute_tolerance`` and ``relative_tolerance`` attributes, respectively. The maximum number of the solver iterations can be set using 
+the ``maximum_iterations`` parameter. Additionally, there is an option to choose linear solver and preconditioning methods that suit better for a particular problem.
+
+The linear solver method can be set with the ``linear_solver`` attribute. The list of available linear solvers can be viewed with: ``print(fenics.list_linear_solver_methods())``.
+
+.. dropdown:: Linear solver methods
+
+    * "bicgstab" - Biconjugate gradient stabilized method
+    * "cg" - Conjugate gradient method
+    * "gmres" - Generalized minimal residual method
+    * "minres" - Minimal residual method
+    * "mumps" - MUMPS (MUltifrontal Massively Parallel Sparse direct Solver)
+    * "petsc" - PETSc built in LU solver
+    * "richardson" - Richardson method 
+    * "superlu" - SuperLU
+    * "superlu_dist" - Parallel SuperLU
+    * "tfqmr" - Transpose-free quasi-minimal residual method
+    * "umfpack" - UMFPACK (Unsymmetric MultiFrontal sparse LU factorization)
+
+The preconditioner can be set with the ``preconditioner`` attribute. The list of available preconditioners can be viewed with: ``print(fenics.list_krylov_solver_preconditioners())``.
+
+.. dropdown:: Preconditioners
+
+    * "amg" - Algebraic multigrid
+    * "hypre_amg" - Hypre algebraic multigrid (BoomerAMG)
+    * "hypre_euclid" - Hypre parallel incomplete LU factorization
+    * "hypre_parasails" - Hypre parallel sparse approximate inverse
+    * "icc" - Incomplete Cholesky factorization
+    * "ilu" - Incomplete LU factorization
+    * "jacobi" - Jacobi iteration 
+    * "petsc_amg" - PETSc algebraic multigrid
+    * "sor" - Successive over-relaxation
+
+Similarly, the Newton solver parameters of :class:`festim.HeatTransferProblem`, :class:`festim.ExtrinsicTrap`, or :class:`festim.NeutronInducedTrap` 
+can be defined upon the need. Here is an example for the heat transfer problem:
+
+.. code-block:: python
+
+    model.T = HeatTransferProblem(
+        transient=True,
+        initial_condition=300,
+        absolute_tolerance=1.0,
+        relative_tolerance=1e-10,
+        maximum_iterations=50,
+        linear_solver="gmres",
+        preconditioner="icc")
+
+
+--------------
+Custom solver
+--------------
+
+For a more flexible control, the built-in Newton solver can be overwritten by a custom solver based on the ``fenics.NewtonSolver`` class. 
+A user-defined Newton solver can be provided after :class:`festim.Simulation.initialise()`:
+
+.. code-block:: python
+
+    class CustomSolver(f.NewtonSolver):
+        def __init__(self):
+            super().__init__()
+
+    custom_solver = CustomSolver()
```

Comment:
Since you're not modifying any of the class functionality you can simply do this no?

```suggestion
    import fenics as f

    custom_solver = f.NewtonSolver()
```

It may be worth showing an example with a modification of the functionality too then

Reply:
I can add both examples: with/without modifications.

What do you think of [this](https://fenicsproject.discourse.group/t/set-krylov-linear-solver-paramters-in-newton-solver/1070/3) example with a modification of the functionality?

Reply:
Or maybe the one that you use for convergence plots?

Reply:
Yeah. Will add.

For this initial attempt, I just didn't want to write something complicated, but rather illustrate the possibility of using the overwritten `fenics.NewtonSolver`.

Reply:
Up to you Vladimir!